domain-models-runtime/src/main/resources/templates/db_scripts/foreign_keys.ftl

```diff
@@ -4,7 +4,21 @@
     <#list table.foreignKeys?filter(key -> key.fieldName??) as key>
       <#if key.tOps.name() == "ADD">
         ALTER TABLE ${myuniversity}_${mymodule}.${table.tableName}
-          ADD COLUMN IF NOT EXISTS ${key.fieldName} UUID REFERENCES ${myuniversity}_${mymodule}.${key.targetTable};
+          ADD COLUMN IF NOT EXISTS ${key.fieldName} UUID;
+        DO $$
+        BEGIN
+          <#-- PostgreSQL doesn't have an "ADD CONSTRAINT IF NOT EXISTS" statement.
```

Comment:
I'm not sure it's any cleaner, but couldn't you also do a DROP CONSTRAINT IF EXISTS, then ADD CONSTRAINT ?

Reply:
oh I see now that someone else made that suggestion in the stackoverflow question, so you probably already considered it.

Reply:
Adding the constraint after it has been dropped requires a check of the constraint condition for all records. This may take long if there are many records, for example in inventory with several million instances, holdings and items that are in foreign key relation.